src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs

```diff
@@ -325,6 +325,37 @@ public void FailsToResolve()
             }
         }
 
+        [Fact]
```

Comment:
Turning this into theory with positive and negative case (2.proj set for matching and mismatching platform, then the expected number of nodes should be 1 and 0 respectively)